service/src/main/java/bio/terra/externalcreds/controllers/OauthApiController.java

```diff
@@ -43,7 +43,20 @@ public ResponseEntity<List<String>> listProviders() {
   public ResponseEntity<LinkInfo> getLink(Provider provider) {
     var samUser = samUserFactory.from(request);
     var linkedAccount = linkedAccountService.getLinkedAccount(samUser.getSubjectId(), provider);
-    return ResponseEntity.of(linkedAccount.map(OpenApiConverters.Output::convert));
+    Optional<LinkInfo> linkInfo = linkedAccount.map(OpenApiConverters.Output::convert);
+    if (linkInfo.isPresent() && provider == Provider.RAS) {
+      var auditLogEventBuilder =
+          new AuditLogEvent.Builder()
+              .provider(provider)
+              .userId(samUser.getSubjectId())
+              .clientIP(request.getRemoteAddr());
+      var era_user_id =
+          providerService.getLinkedEraIdentity(provider, linkedAccount.get(), auditLogEventBuilder);
```

Comment:
I don't know this code very well, but is there a chance of this throwing some kind of error, and if so, would we want to handle it or just let it throw?

Reply:
Oh you're right! This could throw an error if getting the user info from RAS fails...I think it's the connection refused error but I wonder if I should just add a more general try/catch around this line and then log the exception?

Reply:
Ah I see it's defined down below... same question, passed down to that method then maybe